packages/service-library/tests/aiohttp/long_running_tasks/conftest.py

```diff
@@ -0,0 +1,145 @@
+    num_strings: int,
+    sleep_time: float,
+    fail: bool,
+) -> web.Response:
+    generated_strings = []
+    for index in range(num_strings):
+        generated_strings.append(f"{index}")
+        await asyncio.sleep(sleep_time)
+        task_progress.update(message="generated item", percent=index / num_strings)
+        if fail:
+            raise RuntimeError("We were asked to fail!!")
+
+    # NOTE: this code is used just for the sake of not returning the default 200
+    return web.json_response(
+        data={"data": generated_strings}, status=web.HTTPCreated.status_code
+    )
+
+
+@pytest.fixture
+def task_context(faker: Faker) -> TaskContext:
+    return {"user_id": faker.pyint(), "product": faker.pystr()}
+
+
+@pytest.fixture
+def long_running_task_entrypoint() -> str:
+    return "long_running_entrypoint"
+
+
+@pytest.fixture
+def server_routes(
+    task_context: TaskContext, long_running_task_entrypoint: str
+) -> web.RouteTableDef:
+    routes = web.RouteTableDef()
+
+    class _LongTaskQueryParams(BaseModel):
+        num_strings: int
+        sleep_time: float
+        fail: bool = False
+
+    @routes.post("/long_running_task:start", name=long_running_task_entrypoint)
+    async def generate_list_strings(request: web.Request):
+        task_manager = long_running_tasks.server.get_tasks_manager(request.app)
+        query_params = parse_request_query_parameters_as(_LongTaskQueryParams, request)
+        assert task_manager, "task manager is not initiated!"
+        task_name = create_task_name_from_request(request)
+        task_id = long_running_tasks.server.start_task(
+            task_manager,
+            _string_list_task,
+            num_strings=query_params.num_strings,
+            sleep_time=query_params.sleep_time,
+            fail=query_params.fail,
+            task_context=task_context,
+            task_name=task_name,
+        )
+        assert task_id
+        assert task_id.startswith(urllib.parse.quote(task_name, safe=""))
+        return web.json_response(
+            data={"data": task_id},
+            status=web.HTTPAccepted.status_code,
+            dumps=json_dumps,
+        )
+
+    return routes
+
+
+@pytest.fixture
+def start_task(
+    long_running_task_entrypoint,
+) -> Callable[[TestClient], Awaitable[TaskId]]:
+    async def _caller(client: TestClient, **query_kwargs) -> TaskId:
+        assert client.app
+        url = (
+            client.app.router[long_running_task_entrypoint]
+            .url_for()
+            .update_query(num_strings=10, sleep_time=f"{0.2}", **query_kwargs)
+        )
+        resp = await client.post(f"{url}")
+        data, error = await assert_status(resp, web.HTTPAccepted)
+        assert data
+        assert not error
+        task_id = parse_obj_as(long_running_tasks.server.TaskId, data)
+        return task_id
+
+    return _caller
+
+
+@pytest.fixture
+def task_waiter() -> Callable[[TestClient, TaskId, TaskContext], Awaitable[None]]:
```

Comment:
following same naming as previous fixture , i would have called this ``wait_for_long_running_task``

Reply:
here I disagree. I will name it "wait_for_task" cause it waits for any task. it is inside the package long_running_tasks.